### DISCUSSION TITLE
enforce initial velocity

### DISCUSSION THREAD

**Discussion by lyyc199586** [idaholab/moose#26111] [Q&A Modules: Solid mechanics]
In elastodynamic simulations (variable: displacement), is there a way to enforce the initial velocity (not displacement) to some boundary?

**Comment by maxnezdyur** [comment]
Depending on how you set up your simulation, if you are using the `NewmarkVelAux` and a velocity `AuxVariable` then you should be able to set the initial condition on the velocity variable. I think you just need to make sure the `NewmarkVelAux` only runs on `Timestep_End` and not `Linear`.  Option 2, use the `PresetVelocity` BC, and then use controls to turn it off after a small amount of time. It won't be exactly an initial condition but may work in cases were you have some wiggle room with timing.

**Reply by lyyc199586** [reply]
I will try Option 2 first, thanks max!
